### DISCUSSION TITLE
Is it possible to access the Ollama API

### DISCUSSION THREAD

**Discussion by dadofsambonzuki** [Jeffser/Alpaca#334] [Q&A]
There are some VSCode plugins that can access local LLMs. I'm using [Continue](https://marketplace.visualstudio.com/items?itemName=Continue.continue).

Is it possible to access the local Ollama / Alpaca API so that extensions can use?

**Comment by Jeffser** [comment]
Hi yes I just checked and it works!

What you gotta do is go to `~/.var/app/com.jeffser.Alpaca/config/server.json` and change `local_port` to `11434`

Then on the Continue menu go to `Local with Ollama` if it doesn't connect automatically just click the Ollama URL and it will try to connect itself

If you have any other questions feel free to ask!

**Reply by Jeffser** [reply]
Also if you want to close Alpaca and still have the Ollama instance running in the background you can go to Alpaca > Preferences > Run Alpaca In Background

**Reply by dadofsambonzuki** [reply]
Nice! When I change the port number to 11434, it crashes after a few seconds. Changing back (or to another value) fixes it again.

I must have a port clash from a borked install/removal of Ollama, although I don't see anything running on 11434.

EDIT - It does briefly start-up on 11434 before bailing.

**Reply by Jeffser** [reply]
yeah there's probably something else using that port, that's why the default one is different

**Reply by dadofsambonzuki** [reply]
I've got ...
```
127.0.0.1:11435  users:(("ollama",pid=6482,fd=3))                                                       
127.0.0.1:44203  users:(("ollama_llama_se",pid=5350,fd=3)) 
```
The second service continues to run after I close Alpaca. Nothing on 11424 according to ss.

**Reply by Jeffser** [reply]
Just kill those processes and try again, sometimes Ollama keeps running when it isn't supposed to

**Reply by dadofsambonzuki** [reply]
5350 killed.

No change to starting-up on 11434 - i.e. it starts, then aborts.

Nothing obvious in [tmp.log](https://github.com/user-attachments/files/17213968/tmp.log).

What I have found is if I try and start a second time it does load.

**Reply by elifarley** [reply]
> `~/.var/app/com.jeffser.Alpaca/config/server.json` and change `local_port` to `11434`

In my case, I don't see that file:
```shell
ls -l ~/.var/app/com.jeffser.Alpaca/config/¬                                                                                                             
    1 drwx------ 2 user 40 Jan 21 19:59 enchant/¬                                                                                                              
    2 drwx------ 3 user 22 Jan 28 11:10 glib-2.0/¬                                                                                                             
    3 drwx------ 2 user  6 Jan 28 11:10 gtk-4.0/¬                                                                                                              
    4 drwx------ 3 user 17 Jan 21 19:59 ibus/¬                                                                                                                 
    5 drwxr-xr-x 2 user  6 Jan 21 19:59 matplotlib/¬ 
```

**Reply by Jeffser** [reply]
The configuration parameters have been moved to `~/.var/app/com.jeffser.Alpaca/data/alpaca.db`

You'll need to open it with a database viewer for sqlite, there's a bunch of them in flathub

**Reply by elifarley** [reply]
I've inserted a row into table `preferences`:
```sql
insert into preferences (id, value, type) values( 'local_port', '1000', '<class \'int\'>')
```

After that, Alpaca is unable to connect to its Ollama server:

```shell
flatpak run com.jeffser.Alpaca
INFO	[main.py | main] Alpaca version: 3.5.0
INFO	[connection_handler.py | start] Starting Alpaca's Ollama instance...
INFO	[connection_handler.py | start] Started Alpaca's Ollama instance
INFO	[connection_handler.py | start] client version is 0.5.4
ERROR	[model_widget.py | update_local_list] HTTPConnectionPool(host='127.0.0.1', port=1000): Max retries exceeded with url: /api/tags (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x735ddc7baff0>: Failed to establish a new connection: [Errno 111] Connection refused'))
ERROR	[window.py | connection_error] Connection error
INFO	[connection_handler.py | reset] Resetting Alpaca's Ollama instance
INFO	[connection_handler.py | stop] Stopping Alpaca's Ollama instance
INFO	[connection_handler.py | stop] Stopped Alpaca's Ollama instance
INFO	[connection_handler.py | start] Starting Alpaca's Ollama instance...
INFO	[connection_handler.py | start] Started Alpaca's Ollama instance
INFO	[connection_handler.py | start] client version is 0.5.4
INFO	[window.py | show_toast] There was an error with the local Ollama instance, so it has been reset
```

In fact, what I really need is to connect to the Ollama API endpoint, it doesn't matter on which port. I just need to know the port.
I can see in the source code that the [default port is 11435](https://github.com/Jeffser/Alpaca/blob/8653cb68c003c6cc5d1c67714f8701044e42f3b8/src/connection_handler.py#L42), so I don't need to change that.

Trying to connect on that port worked fine:

```shell
curl http://127.0.0.1:11435/
Ollama is running
```

**Comment by ntnsndr** [comment]
Exciting to see that this works. Has anyone tried it with the emacs chatbots, like Ellama or gptel? I have just started tinkering and haven't gotten either working, but I haven't tried all the configuration options yet.
